llvm/lib/Support/CMakeLists.txt

```diff
@@ -295,6 +296,14 @@ add_llvm_component_library(LLVMSupport
   Demangle
   )
 
+# Link icu library if it is an external library.
+if(ICU_FOUND)
+  target_link_libraries(LLVMSupport
+  PRIVATE
+  ${ICU_LIBRARIES}
```

Comment:
I think this means we have to include the ICU license notice somewhere, right?

(https://github.com/unicode-org/icu/blob/main/LICENSE says you have the right to use the software "provided that either (a) this copyright and permission notice appear with all copies of the Data
Files or Software, or (b) this copyright and permission notice appear in associated Documentation.")

Reply:
Thanks for catching! I copied the license text from here https://github.com/unicode-org/icu/blob/main/LICENSE into to CharSet.cpp file. Please let me know if I need to do anything more

Reply:
My understanding of https://discourse.llvm.org/t/rfc-enabling-fexec-charset-support-to-llvm-and-clang-reposting/71512/30 was that the point of only allowing dynamically linked ICU was to avoid the attribution requirement?  Or am I misunderstanding something here?

Reply:
That was my initial understanding. @cor3ntin I think you added similar license text here in the past for Unicode https://github.com/llvm/llvm-project/blob/main/llvm/lib/Support/UnicodeNameToCodepointGenerated.cpp#L17. Would you be able to advise on whether we require the license text for this specific case? Thanks

Reply:
This is the license for Unicode data files which is separate from ICU.

ICU is afaik under MIT license.
My understanding is that we do not need to include the license because we do not ship the software nor do we modify it, as @efriedma-quic  said.

However IANAL so I cannot make a definitive statement :)

Reply:
Ah sorry for the confusion, you're right, we did already figure this out. Sorry, I think there's nothing to do here then -- we're not *requiring* the dynamic link, it's opt-in.

Reply:
Ok thanks, I've reverted adding the license text